cli/src/config.rs

```diff
@@ -162,27 +163,65 @@ impl AsMut<StackedConfig> for RawConfig {
     }
 }
 
+#[derive(Copy, Clone, Debug)]
+struct DeprecatedLocation {
+    old: &'static str,
+    new: &'static str,
+}
+#[derive(Clone, Debug)]
+enum ConfigPathState {
+    New,
+    Exists,
+    Deprecated(DeprecatedLocation),
```

Comment:
nit: perhaps, `Deprecated` can be moved to `ConfigPath { .., deprecated: Option<DeprecatedLocation> }`? The deprecated state is different axis than file/directory existence.

Reply:
It's technically a different axis, but then we'd have:

```rust
#[derive(Clone, Debug)]
pub struct ConfigPath {
    path: PathBuf,
    state: ConfigPathState,
    deprecated: Option<DeprecatedLocation>, // only Some if state == Exists
}
```

which I... don't love. I can do that though.

Reply:
I think it's technically valid to have deprecated "New" path. That said, we won't have to record the deprecated state anyway if we address https://github.com/jj-vcs/jj/pull/6300#discussion_r2059420658.